### ISSUE TITLE
[CheckpointServer] use streaming transfers

### ISSUE THREAD

**Issue by d4l3k** [pytorch-labs/torchft#36]
The CheckpointServer currently uses torch.save/torch.load which requires allocating the entire buffer into memory.

We want to instead use streaming transfers so we minimize the amount of CPU memory required.

It would also be nice to add checksums to these transfers to avoid any data corruption from the network.

The algorithm is described at: https://gist.github.com/d4l3k/b68094d649a076384967788c9b0a5f08

**Comment by d4l3k** [comment]
@Krishn1412 would you be interested in working on this?

**Comment by Krishn1412** [comment]
Sure @d4l3k, I'll work on this

**Comment by Krishn1412** [comment]
Hey @d4l3k , can you take a look at this? https://github.com/pytorch-labs/torchft/pull/54

**Comment by d4l3k** [comment]
@fegin yeah, that's a good point -- we should be able to support DTensor without too much trouble though
